python3Packages.plotly: 5.24.1 -> 6.0.1

WeetHet opened the pull request:
Updates plotly to 6.0.1

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

WeetHet committed:
python3Packages.plotly: add WeetHet to maintainers

WeetHet committed:
python3Packages.plotly: 5.24.1 -> 6.0.1

WeetHet commented on pkgs/development/python-modules/plotly/default.nix:
```diff
@@ -20,60 +22,41 @@
 
   # packages/python/plotly/optional-requirements.txt
   optional-dependencies = {
-    orca = [
-      orca
-      requests
-      psutil
-    ];
```
Seemingly gone, even though there's still an `orca` section on the website. It's not in the `pyproject.toml` and isn't in the `optional-requirements.txt`

Pandapip1 replied:
Orca seems to have been deprecated (and as you've noted, the dependency has been removed), so this seems reasonable.

WeetHet commented on pkgs/development/python-modules/plotly/default.nix:
```diff
@@ -93,27 +76,26 @@ buildPythonPackage rec {
     [
-      # unable to locate orca binary, adding the package does not fix it
-      "plotly/tests/test_orca/"
+      # needs anywidget, but adding that requires building JS files
+      "tests/test_io/test_to_from_json.py"
```
Not sure about this one, feels bad to completely ignore it, but I wasn't able to get it to work

WeetHet committed:
python3Packages.chart-studio: 1.1.0-unstable-2024-07-23 -> 1.1.0-unstable-2025-01-30

WeetHet commented:
A bunch of packages broke after this, so I'm gonna try and address the fallout now

Pandapip1 commented:
This should probably target staging given the number of dependents?

WeetHet commented:
> This should probably target staging given the number of dependents?

I'm rebasing onto it, yeah

-----

> Numpy is seemingly not a propagated input

As it should be, see `optional-dependencies`

WeetHet commented:
Staging has no binary cache, nooo, I don't wanna rebuild stdenv 😢

Pandapip1 commented:
> As it should be, see `optional-dependencies`

Whoops, missed that express was an optional component. Please disregard.

WeetHet committed:
das: relax plotly

WeetHet committed:
python3Packages.niaarm: relax plotly

WeetHet committed:
python3Packages.kmapper: disable tests failing with plotly 6

WeetHet commented:
Broken packages:
- [x] das
- [x] chart-studio
- [x] compressai
- [x] kmapper
- [x] niaarm
- [ ] pymatgen
- [x] scikit-tda
- [ ] sumo

WeetHet committed:
python312Packages.pymatgen: 2025.1.9 -> 2025.4.10

Diff: https://github.com/materialsproject/pymatgen/compare/refs/tags/v2025.1.24...v2025.4.10
Changelog: https://github.com/materialsproject/pymatgen/releases/tag/v2025.4.10

WeetHet commented:
`sumo` and `compressai` seemingly just build? Can't run `nixpkgs-review` since `boehm-gc` seemingly fails to build on staging

Pandapip1 commented:
Might be worth drafting this PR until it's ready for review?

pbsds commented:
You don't have to target staging for <1000 python rebuilds, keep your sanity instead.

Btw, for the future: using `rebase --onto staging..master` to target a common ancestor between master and staging will allow you to build with help from the cache while still being able to target staging in the PR

Pandapip1 commented:
> You don't have to target staging for <500 python rebuilds, keep your sanity instead.

I somehow remember the mandatory staging threshold as 500 packages, and the threshold I'd been personally using was 100. I can no longer seem to find a reference to any threshold at all (maybe it got removed with the infrastructure change?). If you're happy with this PR targeting master, I trust your judgment call on this more than mine.

Pandapip1 commented on pkgs/development/python-modules/plotly/default.nix:
Diff LGTM for the plotly package bump. I'll start my own nixpkgs-review session to see if anything's obviously broken.

WeetHet replied:
Everything is broken, I wasn't able to get even halfway to building plotly since rebasing to staging

WeetHet commented:
Switched back to master to keep my sanity

WeetHet commented:
Rebased on top of latest master, let's see if nixpkgs-review passes

pbsds commented:
It seems 6.1.0 is on `staging` via `python-updates`

pbsds commented:
The fixups however might still be interesting to apply. current location of plotly 6.1.0: https://nixpk.gs/pr-tracker.html?pr=408185
